### PR TITLE
Throw an error if the fetch response returns an error (irrespective o…

### LINKED ISSUE #4100: RemoteConfig.remoteConfig().fetch return .success while the app is offline
### Describe your environment

  * Xcode version: 11.0 (11A420a)
  * Firebase SDK version: 6.10.0
  * Firebase Component: Remote Config
  * Component version: 6.10.0

### Describe the problem
`RemoteConfig.remoteConfig().fetch` returns a `.success` status (`FIRRemoteConfigFetchStatus`) even though I disabled the internet connection.

Note: I never configured defaults via `setDefaults`, and I ran the app for the first time to be sure no previous config was cached.

#### Steps to reproduce:
- Create a new project importing Firebase/RemoteConfig through Cocoapods
- Follow this [guide](https://firebase.google.com/docs/remote-config/use-config-ios) by skipping the step 2 consisting of setting in-app defaults. 
- Disable the internet connection
- Run the app, the remote config is fetched

What happened? How can we make the problem occur?
I got a `.success` status and a `nil` error even though the app is offline and I didn't set any defaults. I expect to get an error instead.

### PR DESCRIPTION
**Pull request by dmandar**
…f status code).
Fixes #4100

### CODE REVIEW

File: FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m [hunk 1]
```diff
@@ -558,6 +558,128 @@ - (void)testFetchConfigsFailed {
                                }];
 }
 
+- (void)testFetchConfigsFailedErrorNoNetwork {
```
Comment (maksymmalyhin) [inline FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: nit: I wonder if it is too big effort to break this test to several methods, e.g. refactor all mocks setup to a separate method that can be called from the for-loop (and may be potentially used by other tests). Also, separate methods for configuring individual mocks may improve readability and code reuse. 

Feel free to consider it as a separate improvements.

File: FirebaseRemoteConfig/Sources/RCNFetch.m [hunk 2]
```diff
@@ -370,6 +370,9 @@ - (void)fetchWithUserProperties:(NSDictionary *)userProperties
           }
         }
         if (statusCode != kRCNFetchResponseHTTPStatusCodeOK) {
+          // Update metadata about fetch failure.
+          [strongSelf->_settings updateMetadataWithFetchSuccessStatus:NO];
```
Comment (maksymmalyhin) [inline FirebaseRemoteConfig/Sources/RCNFetch.m]: Looks like by this line the method is already called at line 362. Do we need to call this method 2nd time?

File: FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m (hunk 1 above)
Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: sg. Other tests have the mocks initialized in the setup section. It is only the testFetchConfigsFailed and now this test that need a different response code back. But yes, makes sense to refactor..will make that a separate change.

File: FirebaseRemoteConfig/Sources/RCNFetch.m (hunk 2 above)
Comment (dmandar) [reply FirebaseRemoteConfig/Sources/RCNFetch.m]: right. Deleted.

File: FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m (hunk 1 above)
Comment (chliangGoogle) [reply FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: maybe add a todo here.

Comment (dmandar) [reply FirebaseRemoteConfig/Tests/Unit/RCNRemoteConfigTest.m]: Added.

**Pull request merged**